web/src/components/AttributeRow/AttributeCheck.tsx

```diff
@@ -0,0 +1,54 @@
+import {Dropdown, Menu} from 'antd';
+
+import {useAssertionForm} from 'components/AssertionForm/AssertionFormProvider';
+import {IResult} from 'components/SpanDetail/SpanDetail';
+import * as S from './AttributeRow.styled';
+
+interface IProps {
+  items: IResult[];
+  type: 'error' | 'success';
+}
+
+const AttributeCheck = ({items, type}: IProps) => {
+  const {setIsCollapsed} = useAssertionForm();
+
+  const handleOnClick = (id: string) => {
+    setIsCollapsed(true);
+    const resultListElement = document.getElementById(`assertion-${id}`);
+    const offsetTop = resultListElement?.offsetTop ?? 0;
+    const assertionsContainer = document.getElementById('assertions-container');
+    if (assertionsContainer) {
+      assertionsContainer.scrollTop = offsetTop;
+    }
+  };
+
+  const menuLayout = (
+    <Menu
+      items={items.map(item => ({
+        key: item.id,
+        label: item.label,
+      }))}
+      onClick={({key}) => handleOnClick(key)}
+    />
+  );
+
+  if (items.length === 1) {
+    return (
+      <div onClick={() => handleOnClick(items[0].id)}>
+        <S.CustomBadge status={type} text={items.length} />
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      <Dropdown overlay={menuLayout} placement="bottomLeft" trigger={['click']}>
+        <div>
```

Comment:
Is this extra div necessary?

Reply:
Necessary for the Dropdown. A Badge cannot be a direct child of Dropdown because it doesn't have an onClick prop.